src/Microsoft.VisualStudio.Services.Agent/ConfigurationStore.cs

```diff
@@ -29,6 +29,9 @@ public sealed class AgentSettings
         [DataMember(EmitDefaultValue = false)]
         public string NotificationSocketAddress { get; set; }
 
+        [DataMember(EmitDefaultValue = false)]
+        public bool IsProviderAgent { get; set; }
```

Comment:
No

Reply:
And this should be an env var since it's something to influence the behavior that is not a typical valid config option for an end user - it's something environmental